core/lib/l1_contract_interface/src/i_executor/methods/commit_batches.rs

```diff
@@ -1,25 +1,46 @@
 use zksync_types::{commitment::L1BatchWithMetadata, ethabi::Token, pubdata_da::PubdataDA};
 
 use crate::{
-    i_executor::structures::{CommitBatchInfo, StoredBatchInfo},
+    i_executor::structures::{CommitBatchInfoRollup, CommitBatchInfoValidium, StoredBatchInfo},
     Tokenizable, Tokenize,
 };
 
-/// Input required to encode `commitBatches` call.
+/// Input required to encode `commitBatches` call for a contract running in rollup mode.
 #[derive(Debug, Clone)]
-pub struct CommitBatches {
+pub struct CommitBatchesRollup {
     pub last_committed_l1_batch: L1BatchWithMetadata,
     pub l1_batches: Vec<L1BatchWithMetadata>,
     pub pubdata_da: PubdataDA,
 }
 
-impl Tokenize for CommitBatches {
+impl Tokenize for CommitBatchesRollup {
     fn into_tokens(self) -> Vec<Token> {
         let stored_batch_info = StoredBatchInfo(&self.last_committed_l1_batch).into_token();
         let l1_batches_to_commit = self
             .l1_batches
             .iter()
-            .map(|batch| CommitBatchInfo::new(batch, self.pubdata_da).into_token())
+            .map(|batch| CommitBatchInfoRollup::new(batch, self.pubdata_da).into_token())
+            .collect();
+
+        vec![stored_batch_info, Token::Array(l1_batches_to_commit)]
+    }
+}
+
+/// Input required to encode `commitBatches` call for a contract running in validium mode.
+#[derive(Debug, Clone)]
+pub struct CommitBatchesValidium {
```

Comment:
drive by comment -- this seems to be very similar to commitBatchesRollup. I'd have personally have a single struct, with an enum (that would later hold the things that are different between the two).

Reply:
This was done per our request, this is intentional.